File: pruebaretrofitjava/macasado-pruebaretrofitjava-3a64fe4428ea/pruebaRetrofitJava/src/pruebaRetrofitJava/Libro.java
package pruebaRetrofitJava;

public class Libro {
	private int codigo;
	private String titulo;
	private String numpag;

	public Libro(int codigo, String titulo, String numpag) {
		this.codigo = codigo;
		this.titulo = titulo;
		this.numpag = numpag;
	}

	public Libro(String titulo, String numpag) {
		this.titulo = titulo;
		this.numpag = numpag;
	}
	
	public int getId() {
		return codigo;
	}
	public void setId(int codigo) {
		this.codigo = codigo;
	}
	public String getNumpag() {
		return numpag;
	}
	public void setNumpag(String numpag) {
		this.numpag = numpag;
	}
	public String getTitulo() {
		return titulo;
	}
	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}
}

File: Examen1Trimestre/clases/Libro.php
<?php


class Libro implements JsonSerializable
{
    private $codigo;
    private $titulo;

    public function __construct($codigo,$titulo)
    {
        $this->codigo=$codigo;
        $this->titulo=$titulo;
    }

    //es para el json
    function jsonSerialize()
    {
        return array(
            'titulo' => $this->titulo,
            'codigo' => $this->codigo
        );
    }

    public function __sleep(){
        return array('titulo' , 'codigo' , 'numpag' );
    }

    public function getTitulo()
    {
        return $this->titulo;
    }

    /**
     * @param mixed $titulo
     */
    public function setTitulo($titulo)
    {
        $this->titulo = $titulo;
    }

    public function getCodigo()
    {
        return $this->codigo;
    }

    /**
     * @param mixed $codigo
     */
    public function setCodigo($codigo)
    {
        $this->codigo = $codigo;
    }
}